sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpressionJsonConverter.cs

```diff
@@ -11,32 +15,157 @@ internal class DataFactoryExpressionJsonConverter : JsonConverter<object>
     {
         public override bool CanConvert(Type typeToConvert)
         {
-            return (typeToConvert == typeof(DataFactoryExpression<string>) ||
-                typeToConvert == typeof(DataFactoryExpression<int>) ||
-                typeToConvert == typeof(DataFactoryExpression<double>) ||
-                typeToConvert == typeof(DataFactoryExpression<Array>) ||
-                typeToConvert == typeof(DataFactoryExpression<bool>));
+            return typeToConvert == typeof(DataFactoryExpression<string>) ||
+                    typeToConvert == typeof(DataFactoryExpression<int>) ||
+                    typeToConvert == typeof(DataFactoryExpression<double>) ||
+                    typeToConvert == typeof(DataFactoryExpression<Array>) ||
+                    typeToConvert == typeof(DataFactoryExpression<bool>) ||
+                   IsGenericDataFactoryList(typeToConvert);
         }
 
         public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
             using var document = JsonDocument.ParseValue(ref reader);
             if (typeToConvert == typeof(DataFactoryExpression<string>))
-                return DataFactoryExpression<string>.DeserializeDataFactoryExpression(document.RootElement);
+                return Deserialize<string>(document.RootElement);
             if (typeToConvert == typeof(DataFactoryExpression<int>))
-                return DataFactoryExpression<int>.DeserializeDataFactoryExpression(document.RootElement);
+                return Deserialize<int>(document.RootElement);
             if (typeToConvert == typeof(DataFactoryExpression<double>))
-                return DataFactoryExpression<double>.DeserializeDataFactoryExpression(document.RootElement);
+                return Deserialize<double>(document.RootElement);
             if (typeToConvert == typeof(DataFactoryExpression<Array>))
-                return DataFactoryExpression<Array>.DeserializeDataFactoryExpression(document.RootElement);
+                return Deserialize<Array>(document.RootElement);
             if (typeToConvert == typeof(DataFactoryExpression<bool>))
-                return DataFactoryExpression<bool>.DeserializeDataFactoryExpression(document.RootElement);
+                return Deserialize<bool>(document.RootElement);
+            if (IsGenericDataFactoryList(typeToConvert))
+            {
+                var methodInfo = typeof(DataFactoryExpressionJsonConverter)
```

Comment:
maybe share this this code with the Write method only diff seems to the be the name of the method